Source/WebKit/WPEPlatform/wpe/WPEView.cpp

```diff
@@ -237,6 +250,33 @@ static void wpe_view_class_init(WPEViewClass* viewClass)
             WPE_TYPE_MONITOR,
             WEBKIT_PARAM_READABLE);
 
+    /**
+     * WPEView:visible:
+     *
+     * Whether the view should be visible or not.
+     * The view might not be mapped even if it's set as visible,
+     * for example when it's minimized.
+     */
+    sObjProperties[PROP_VISIBLE] =
+        g_param_spec_boolean(
+            "visible",
+            nullptr, nullptr,
+            TRUE,
+            WEBKIT_PARAM_READWRITE);
+
+    /**
+     * WPEView:mapped:
+     *
+     * Whether the view is mapped or not. A view is mapped when #WPEView:visible
```

Comment:
I'm confused. For PROP_VISIBLE you mention, a view might not be mapped even if it's set as visible (e.g. for minimized windows). PROP_MAPPED says a view is mapped if it's visible, and not hidden (e.g. minimized). That's a contradiction.

Reply:
I don't see the contradiction, if a view toplevel is minimized, in a different workspace, the screen is locked, etc. the view is never mapped, no matter what the visibility property says. But to be mapped it also needs to be visible.

Reply:
I think the wording was just difficult to interpret - much clearer now.